spec/models/tower_spec.rb

```diff
@@ -1,5 +1,95 @@
 require 'rails_helper'
 
 RSpec.describe Tower, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  describe '#valid?' do
+    it 'false when name is blank' do
+      tower = Tower.new name: ''
+
+      tower.valid?
+
+      expect(tower.errors.full_messages
+        .include?('Nome não pode ficar em branco')).to be true
+    end
+
+    it 'false when floor quantity is blank' do
+      tower = Tower.new floor_quantity: ''
+
+      tower.valid?
+
+      expect(tower.errors.full_messages
+        .include?('Quantidade de Andares não pode ficar em branco')).to be true
+    end
+
+    it 'false when units per floor is blank' do
+      tower = Tower.new units_per_floor: ''
+
+      tower.valid?
+
+      expect(tower.errors.full_messages
+        .include?('Apartamentos por Andar não pode ficar em branco')).to be true
+    end
+
+    it 'false when condo is missing' do
+      tower = Tower.new
+
+      tower.valid?
+
+      expect(tower.errors.full_messages
+        .include?('Condomínio é obrigatório(a)')).to be true
+    end
+
+    it "false when floor quantity isn't a number" do
+      tower = Tower.new floor_quantity: 'ten'
+
+      tower.valid?
+
+      expect(tower.errors.full_messages
+        .include?('Quantidade de Andares não é um número')).to be true
+    end
+
+    it "false when units per floor isn't a number" do
+      tower = Tower.new units_per_floor: 'five'
+
+      tower.valid?
+
+      expect(tower.errors.full_messages
+        .include?('Apartamentos por Andar não é um número')).to be true
+    end
+
+    it 'true when floor quantity is a positive number' do
+      tower = Tower.new floor_quantity: 0
+
+      tower.valid?
+
+      expect(tower.errors.full_messages
+        .include?('Quantidade de Andares deve ser maior ou igual a 0')).to be false
+    end
+
+    it "false when floor quantity isn't a positive number" do
+      tower = Tower.new floor_quantity: -1
+
+      tower.valid?
+
+      expect(tower.errors.full_messages
+        .include?('Quantidade de Andares deve ser maior ou igual a 0')).to be true
+    end
+
+    it 'true when units per floor is a positive number' do
+      tower = Tower.new units_per_floor: 0
+
+      tower.valid?
+
+      expect(tower.errors.full_messages
+        .include?('Apartamentos por Andar deve ser maior ou igual a 0')).to be false
+    end
+
+    it "false when units per floor isn't a positive number" do
+      tower = Tower.new units_per_floor: -1
+
+      tower.valid?
+
+      expect(tower.errors.full_messages
+        .include?('Apartamentos por Andar deve ser maior ou igual a 0')).to be true
+    end
```

Comment:
Esse comentário é mais sobre preferência do que uma regra ou sugestão, mas nota como toda descrição começa com verdadeiro ou falso? Isso é necessário? Todo teste poderia apenas descrever o que ele valida, por exemplo:

- "units per floor must be positive"
- "units per floor must be a number"

Além disso, acho que alguns testes poderiam ser feitos de uma vez só, por exemplo:

```ruby
#considerando que as factories estão corretamente configuradas

      zero_tower = create(:tower, units_per_floor: 0)
      negative_tower = create(:tower, units_per_floor: -1)

      expec(zero_tower).to be_valid
      expect(negative_tower).not_bo be_valid
      expect(negative_tower.errors.full_messages).to include('Apartamentos por Andar deve ser maior ou igual a 0')
```

Não tem um jeito "certo" de fazer esses testes, mas existem formas de deixar o conjunto de testes mais enxuto.

Reply:
Acabamos de corrigir isso nesse último push, além de termos únido alguns testes agora estabelecemos um padrão para a validação, utilizando de "be_valid" ao invés de "valid?"